### PR TITLE
split-object Reinaldo 

### PR DESCRIPTION
**Pull request by perezrei**
My solution for split-objects

Before the reviewer merges this PR to main/master, all of these boxes must be checked. Only the reviewer can check boxes.

## Behavior

> What the function does - inputs and outputs

- **The Function**
  - [x] The function's name is clear and helpful
  - [x] The parameter names are clear and helpful
  - [x] Default parameters are used if possible
- **JSDoc Comment**
  - [x] The docstring is well-formatted
  - [x] There are no spelling mistakes
  - [x] The description of the function's behavior is clear
  - [x] All parameters are documented with types
  - [x] Documented parameter names match the function's parameter names
  - [x] Parameters have a good description
  - [x] If default parameters are used, they are the correct type
  - [x] The return value is documented with a type
  - [x] The return value has a clear description
- **README**
  - [x] The documentation is rendered into the README
  - [x] Docs were generated just before pushing
  - [x] Docs in the README matches the JSDoc comment
  - [x] The README contains at least one use case
- **Unit Tests**
  - [x] Unit tests are present
  - [ ] There are at least 8 test cases
  - [x] All `describe`s and `it`s have clear and helpful names
  - [x] All test names actually describe what is being tested
  - [ ] Tests are grouped logically using `describe`
  - [x] There is only one `expect` inside every `it`
  - [x] There is no logic (loops, conditionals, ...) in the test cases
  - [ ] Side-effects are tested, if necessary
  - [ ] Tests were run just before pushing
  - [ ] All tests pass
  - [ ] Tricky edge cases are tested (extra credit)

## Strategy

> The logic behind the solution

- **The Function**
  - [x] Variables have names that help understand the strategy
  - [x] Spacing and formatting help to understand the strategy
  - [x] The code is written as simply and clearly as possible
  - [x] Use short, clear comments to describe the strategy
- **README**
  - [x] There is an explanation of the solution's strategy in the README
  - [x] The explanation is clear and understandable
  - [x] The explanation actually matches the function that was written

## Implementation

> The code used to write the solution

- **Formatting**
  - [ ] The code was formatted just before pushing
  - [ ] No errors occurred while formatting the code
- **Linting**
  - [ ] There should be no linting errors or warnings
  - [ ] Spelling should be correct in all code, comments and docstrings
  - [ ] The code was linted just before pushing
- **The solution file**
  - [x] Has a JSDoc comment
  - [x] Has an exported function
  - [x] Does no other code or comments outside of the function body (you can use `sandbox.js` for saving extra ideas)
- **The function body _Should_ ...**
  - [x] Be well and consistently formatted
  - [x] Have clear and helpful variable names
  - [x] Declare variables in the smallest scope possible
  - [x] Be a simple and readable implementation
- **The function body should _Not_ ...**
  - [x] Have extra comments that are not about the solution's strategy
  - [x] Have any user interactions (`prompt`, `alert`, `confirm`)
  - [x] Have any `console` calls (`.log`, `.assert`, ...)
  - [x] Have any `debugger` statements

### CODE REVIEW

**Comment by perezrei** [comment]
---
title: Code Review: Split-Object
---



Before the reviewer merges this PR to main/master, all of these boxes must be checked. Only the reviewer can check boxes.

## Behavior

> What the function does - inputs and outputs

- **The Function**
  - [ ] The function's name is clear and helpful
  - [ ] The parameter names are clear and helpful
  - [ ] Default parameters are used if possible
- **JSDoc Comment**
  - [ ] The docstring is well-formatted
  - [ ] There are no spelling mistakes
  - [ ] The description of the function's behavior is clear
  - [ ] All parameters are documented with types
  - [ ] Documented parameter names match the function's parameter names
  - [ ] Parameters have a good description
  - [ ] If default parameters are used, they are the correct type
  - [ ] The return value is documented with a type
  - [ ] The return value has a clear description
- **README**
  - [ ] The documentation is rendered into the README
  - [ ] Docs were generated just before pushing
  - [ ] Docs in the README matches the JSDoc comment
  - [ ] The README contains at least one use case
- **Unit Tests**
  - [ ] Unit tests are present
  - [ ] There are at least 8 test cases
  - [ ] All `describe`s and `it`s have clear and helpful names
  - [ ] All test names actually describe what is being tested
  - [ ] Tests are grouped logically using `describe`
  - [ ] There is only one `expect` inside every `it`
  - [ ] There is no logic (loops, conditionals, ...) in the test cases
  - [ ] Side-effects are tested, if necessary
  - [ ] Tests were run just before pushing
  - [ ] All tests pass
  - [ ] Tricky edge cases are tested (extra credit)

## Strategy

> The logic behind the solution

- **The Function**
  - [ ] Variables have names that help understand the strategy
  - [ ] Spacing and formatting help to understand the strategy
  - [ ] The code is written as simply and clearly as possible
  - [ ] Use short, clear comments to describe the strategy
- **README**
  - [ ] There is an explanation of the solution's strategy in the README
  - [ ] The explanation is clear and understandable
  - [ ] The explanation actually matches the function that was written

## Implementation

> The code used to write the solution

- **Formatting**
  - [ ] The code was formatted just before pushing
  - [ ] No errors occurred while formatting the code
- **Linting**
  - [ ] There should be no linting errors or warnings
  - [ ] Spelling should be correct in all code, comments and docstrings
  - [ ] The code was linted just before pushing
- **The solution file**
  - [ ] Has a JSDoc comment
  - [x] Has an exported function
  - [x] Does no other code or comments outside of the function body (you can use `sandbox.js` for saving extra ideas)
- **The function body _Should_ ...**
  - [x] Be well and consistently formatted
  - [x] Have clear and helpful variable names
  - [x] Declare variables in the smallest scope possible
  - [x] Be a simple and readable implementation
- **The function body should _Not_ ...**
  - [x] Have extra comments that are not about the solution's strategy
  - [x] Have any user interactions (`prompt`, `alert`, `confirm`)
  - [x] Have any `console` calls (`.log`, `.assert`, ...)
  - [x] Have any `debugger` statements

File: src/split-object/split-object-reinaldo/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+---
+
+## Strategy
+
+To return an array of objects from one object using built-in .js methods Objects.entries() and .map().
+
+## Implementation
+
+The Object.values() method returns an array of a given object's own enumerable property values.
```
Comment (JTLiberona) [inline src/split-object/split-object-reinaldo/README.md]: You used Object.entries() instead of Object.values().

File: src/split-object/split-object-reinaldo/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+The Map object holds key-value pairs and remembers the original insertion order of the keys.
+
+## Use Cases
```
Comment (JTLiberona) [inline src/split-object/split-object-reinaldo/README.md:15]: You should propose a use for it :)

File: src/split-object/split-object-reinaldo/split-object-reinaldo.spec.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      { age: 25 },
+      { tall: true },
+      { userName: 'sbor' },
+    ]);
+  });
+  it('testing with an empty object', () => {
+    const actual = splitObjectReinaldo({});
+    expect(actual).toEqual([]);
+  });
+});
```
Comment (JTLiberona) [inline src/split-object/split-object-reinaldo/split-object-reinaldo.spec.js:77]: 1. You have just 3 test cases and there should be at least 8 test cases.
2. The test cases must be grouped logically using describe. (That is something that could be improved too).
2.1. You need to explicit the side effects in the unit test. ;) Side-effects like: **returns a new array** and **the argument object is not modified**.

File: docs/README.md [hunk 4]
```diff
@@ -21,6 +21,8 @@
   - example-built-in-sort
     - [sort-numbers.js](#srcsort-numbersexample-built-in-sortsort-numbersjs)
 - split-object
+  - split-object-reinaldo
```
Comment (JTLiberona) [inline docs/README.md:24]: Creo que estos intervalos 24-25 y 91-134 deberian estar en el README de tu carpeta :D
Asi se complementa mejor la informacion que tienes en el README de tu solucion y este que esta por default queda limpio.

**Review by JTLiberona — CHANGES_REQUESTED** [review]
I checked your solution Reinaldo and I like it. It is easy to read and understand.

Please check my review(The comments below) and the boxes of the checklist.
Also your branch has a conflict, so take a look on it so after you finish the last things to do I can merge it.

File: src/split-object/split-object-reinaldo/README.md (hunk 1 above)
Comment (perezrei) [reply src/split-object/split-object-reinaldo/README.md]: Oh yes, my mistake,  thanks JT

File: src/split-object/split-object-reinaldo/README.md (hunk 2 above)
Comment (perezrei) [reply src/split-object/split-object-reinaldo/README.md:15]: Ok

File: src/split-object/split-object-reinaldo/split-object-reinaldo.spec.js (hunk 3 above)
Comment (perezrei) [reply src/split-object/split-object-reinaldo/split-object-reinaldo.spec.js:77]: Ok, done the 8 tests
